tests/elemental3/configure_network.pm

```diff
@@ -0,0 +1,59 @@
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Preparing static IPs and hostnames on multi-nodes.
+#
+# Based on code written by Pavel Dostal <pdostal@suse.cz>
+# Maintainer: unified-core@suse.com, ldevulder@suse.com
+
+use base qw(opensusebasetest);
+use testapi;
+use lockapi;
+use mm_network qw(configure_hostname setup_static_mm_network);
+use serial_terminal qw(select_serial_terminal);
+
+sub run {
+    my $hostname = get_required_var('HOSTNAME');
+    my $is_server = ($hostname =~ /master/);
+    my $local_index;
+
+    # Set default root password
+    $testapi::password = get_required_var('TEST_PASSWORD') unless ($is_server);
+
+    # No GUI, easier and quicker to use the serial console
+    select_serial_terminal();
+
+    # Do not use external DNS for our internal hostnames
+    assert_script_run('echo "10.0.2.100 master" >> /etc/hosts');
+    foreach (split(/,/, get_required_var('NODES_LIST'))) {
+        $_ =~ m/node(.*)/;
```

Comment:
I believe one of the nodes can be named simply `master`. If so, it will not match this regexp and leave `undef` in the `$index` assignment below. Better to shield this with an `if` or `unless` so you're sure you get what's expected. I also think it's better to match to `/node(.+)/` unless a hostname without an index (ex. simply `node`) is also valid.

For example:
```
    foreach (split(/,/, get_required_var('NODES_LIST'))) {
        next unless ($_ =~ m/node(.+)/);
        my $index = $1;
        ...
```

Reply:
Yes maybe, I will give it a try, thanks.

Reply:
In fact I'm not sure to follow you.

> I believe one of the nodes can be named simply `master`.

You mean having something like `NODE_LISTS: master,node01,node02,...`?

What I want is to have the same entries added in `/etc/hosts` (I don't want to configure and manage DHCP/DNS on a support server) and doing this way avoid having to copy the files on all VMs. So I don't see why `next unless ...` will be helpful in this case (as well as adding `master` in the list).

The whole test is composed of a `master` VM with a simple MicroOS on it that will execute the `distros-test-framework` suite on a Kubernetes cluster composed of `nodeXX` (only one at the beginning, more to come).